src/query.rs

```diff
@@ -122,6 +122,12 @@ pub struct SemverQuery {
     /// a human-readable description of the specific semver violation that was discovered.
     #[serde(default)]
     pub(crate) per_result_error_template: Option<String>,
+
+    /// Optional data to create witness code for query output.  See the [`Witness`] struct for
+    /// more information.  Witness implements `Default` for a query that doesn't generate any
+    /// witness code.
+    #[serde(default)]
+    pub witness: Witness,
```

Comment:
Couldn't we keep `serde(default)` and make this `Option<Witness>`. Otherwise we might end up in a weird situation where we have no hint text (so nothing to show the user) even though we can generate a full witness example. We'd like weird states like that to be unrepresentable.

I'd make this `Option<Witness>` then make the hint required but the other fields inside still be `Option`.